### ISSUE TITLE
Iris Devices into Home Assistant

### ISSUE THREAD

**Issue by rrubin0** [jamesleesaunders/PyAlertMe#38]
Hi James,

My name is Rick and I recently saw on Dave's (desert-home) site that you're actively doing some work on the AlertMe zigbee devices. I was excited to see this, as I've been trying to migrate off of Lowe's system for a while now and do not wish to abandon all of the devices that I've acquired. 

Finding your project is almost perfect timing, as I'd like to introduce you to a community of users who may be able to help integrate your work into a component for Home Assistant.

If you've not heard of [Home Assistant](https://home-assistant.io/) yet, I think you will love it.
The system is designed to run on RaspberryPi and is built on Python3. The community has already developed over 650 [components ](https://home-assistant.io/components/)for integration and it has been a lot of fun too; great community.

Anyway, the reason I think you will be excited to get involved is that in home assistant you can write your own custom components and eventually merge them into the build. Another user has recently gotten the USB Z-wave plus/zigbee dongle [HUSBZB-1](https://www.amazon.com/GoControl-Linear-HUSBZB-1/dp/B01GJ826F8) working with zigbee devices. I purchased this device, but found that it was limited in that I couldn't get it to communicate eith Iris gen1 devices. When I found your project, I thought that you'd be interested in the efforts to get zigbee up and running in home assistant.

At the core of the zigbee component is a github project called [bellows](https://github.com/rcloran/bellows) which looks promising.

I had reached out to Dave a while back, who also seemed interested in Home Assistant, but he too was a bit too busy. I was very happy to see that you've picked up some of this work! I do not have an XBee to test your project out, but am wondering if it could work on the HUSBZB-1. I'll need to investigate, as I have just found your work this morning...but do hope you will give home assistant a look.

You can find me in the gitter [chatroom](https://gitter.im/home-assistant/) for home assistant or email me if you have questions at rixreply@cox.net

Thanks!

**Comment by jamesleesaunders** [comment]
Hi @rrubin0 

Thanks for getting in contact. I am glad you found this project via Desert Home. Your case is exactly why I developed this package - I expect there are probably thousands of people who have older Iris / AlertMe / British Gas systems who, when time is called and they soon pull the plug on the AlertMe backend will be lumbered with lots of 'useless' sensors, plugs etc of little use. So, I would like to see this package used by anyone with a Pi who wants to swap out their old hubs for a Pi and then give them to option (via Python) to integrate them into their own HA projects.

I am more than happy (screaming out even) for people who would like to collaborate with me on the PyAlertMe project. At the moment I am still 'tinkering' with it really so have not really publicised it (in part because I am still learning Python!). But if and when I get it to a more finalised state I will try to publicise it more. So yes please! If you or anyone you know could help me I will bite your hand off!

I had a quick look at Home Assistant and it does indeed look very exciting - I am impresses with its huge list of list of integrations with other HA systems. I need to look into it a bit more but yes I think PyAlertMe could be a good candidate to integrate into it (I couldn't tell you how yet though). 
I also need to take a peek at Bellows.

I am not sure PyAlertMe will work with HUSBZB-1 stick as, at the moment, it relies on the [Python XBee](https://pypi.python.org/pypi/XBee) library and expects a connected XBee. But I would not say never.

It would probably be a good move to introduce myself on the Home Assistant chat-room - so I may catch you over there!

Thanks so much for getting in touch, I truly believe PyAlertMe could be an extremely useful library (but I am still learning so need some help/guidance on how to make it useful).

All the best,
Jim

**Comment by rrubin0** [comment]
Excellent, Jim - Just what I wanted to hear! - No worries on the python skills; I don't code much at all myself. I can get you in contact with some folks who can tho. I just received a "how's testing going?" response to a [recent forum post](https://community.home-assistant.io/t/lowes-iris-version-1-or-2-hub-in-home-assistant/3781/11), so I will link to your project there and see if we can recruit others to assist in some pull requests to get your work merged with the HUSBZB stick perhaps. 

If nothing else, I have rooted my Gen-1 Iris hub and have a spare G1/2 hub to play with, so using the hub in home assistant would be a win in my book too. Many folks are using the Smart Things, Wink, Vera hubs with home assistant too. I've contacted the one Iris guru who wrote the web interface, and he has joined the forum as you will see. Have not seen much from him, but he could be a resource. Additionally, I've got a contact who is a Lowe's Iris tech support rep who is in the community. He could also prove useful.
